### ISSUE TITLE
[Classic] Prospector Remtravel isn't attacked by monsters during Escort quest The Absent Minded Prospector

### ISSUE THREAD

**Issue by Trystanosaurus** [cmangos/issues#1659]
**Current behavior**: 
When you do the escort the prospector isn't attacked by the golems. He seems to follow the path fine and says all his lines he just doesn't get attacked by the golems.

Also no troggs spawn during the event

He's faction 55 when you're not escorting him and during the escort he's faction 10.

**Expected behavior**: 
The prospector should be attacked by the golems. He's supposed to walk on by if the players are fighting the golems but he should get attacked and stop to fight.

At three points in the event troggs should spawn, they seem to be in his script in darkshore.cpp but didn't appear in game.

**Steps to reproduce**:

1. Add the first Absent Minded prospector quest .quest add 729
2. Go to prospector remtravel .go creature id 2917
3. Hand in the quest and accept the escort follow up
4. The prospector doesn't get attacked by the golems and no troggs appear.

**Client version**: 1.12.1

**Commit hash**: ca5ef78

**Database version**: b5e2f39

**Operating system**: Windows 10

[//]: # (If this is a crash report, include the crashlog from a debug build with https://gist.github.com/)

**Comment by killerwife** [comment]
Check please if he has unit flags 512 removed on start of script. (or if he has them to begin with)I have been slowly fixing these on tbc, but ofc I havent checked all vanilla scripts. @Mantislord check the flag in our sources too.

**Comment by katz1** [comment]
This seems to happen in every escort quest, at least every one I did in the past 3 months.
The escorted NPC is not attacked by anything be it a normal wandering creature of the zone or creatures spawned by the quest's script (to ambush).

Just to give a few examples:
https://vanilla-twinhead.twinstar.cz/?quest=836
https://vanilla-twinhead.twinstar.cz/?quest=2767

**Comment by killerwife** [comment]
Yes, its due to the flags i mentioned and reworked relations. When going over sniffs, you will notice the 512 flag is manipulated more than in our vanilla scripts.

**Comment by katz1** [comment]
Prospector Remtravel and the robot chickens does have unit flag 512 in the DB, and I don't see any UNIT_FLAG_IMMUNE_TO_NPC related things in the script (if I am checking at the right places).
It is probably as you expected. I will try checking more quests I remember having this issue later.

The troggs not spawning is a separate issue, ambushers did spawn in the other quests I did.

**Comment by katz1** [comment]
Here is an incomplete list of quests where unit flag 512 should be removed from the escorted NPC when the quest starts:

https://vanilla-twinhead.twinstar.cz/?quest=648
https://vanilla-twinhead.twinstar.cz/?quest=729
https://vanilla-twinhead.twinstar.cz/?quest=836
https://vanilla-twinhead.twinstar.cz/?quest=976
https://vanilla-twinhead.twinstar.cz/?quest=2767
https://vanilla-twinhead.twinstar.cz/?quest=3367

I noticed that the defias traitor in The Defias Brotherhood escort quest does not have unit flag 512 intially, I wonder if this is intentional (https://vanilla-twinhead.twinstar.cz/?quest=155).

**Comment by killerwife** [comment]
Resolved some time ago with TEMPFACTION_TOGGLE_IMMUNE_TO_NPC
